services/scheduler/src/main/java/org/mule/service/scheduler/internal/executor/WaitPolicy.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.scheduler.internal.executor;
+
+import static java.lang.String.format;
+
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A handler for unexecutable tasks that waits until the task can be submitted for execution or times out. Generously snipped from
+ * the jsr166 repository at:
+ * <a href="http://gee.cs.oswego.edu/cgi-bin/viewcvs.cgi/jsr166/src/main/java/util/concurrent/ThreadPoolExecutor.java"></a>.
+ */
+// @Immutable
+public class WaitPolicy implements RejectedExecutionHandler {
```

Comment:
Do we need to two copies of this, what is the difference? Maybe make this clear in javadoc.